docs/guide/search/extend_solr.md

```diff
@@ -0,0 +1,542 @@
+which are children of the webinar. 
+The field mapper could then look like this:
+
+```php
+ <?php
+
+namespace My\WebinarApp;
+
+use EzSystems\EzPlatformSolrSearchEngine\FieldMapper\ContentFieldMapper;
+use eZ\Publish\SPI\Persistence\Content\Handler as ContentHandler;
+use eZ\Publish\SPI\Persistence\Content\Location\Handler as LocationHandler;
+use eZ\Publish\SPI\Persistence\Content;
+use eZ\Publish\SPI\Search;
+
+class WebinarEventTitleFulltextFieldMapper extends ContentFieldMapper
+{
+    /**
+     * @var \eZ\Publish\SPI\Persistence\Content\Type\Handler
+     */
+    protected $contentHandler;
+
+    /**
+     * @var \eZ\Publish\SPI\Persistence\Content\Location\Handler
+     */
+    protected $locationHandler;
+
+    /**
+     * @param \eZ\Publish\SPI\Persistence\Content\Handler $contentHandler
+     * @param \eZ\Publish\SPI\Persistence\Content\Location\Handler $locationHandler
+     */
+    public function __construct(
+        ContentHandler $contentHandler,
+        LocationHandler $locationHandler
+    ) {
+        $this->contentHandler = $contentHandler;
+        $this->locationHandler = $locationHandler;
+    }
+
+    public function accept(Content $content)
+    {
+        // ContentType with ID 42 is webinar event
+        return $content->versionInfo->contentInfo->contentTypeId == 42;
+    }
+
+    public function mapFields(Content $content)
+    {
+        $mainLocationId = $content->versionInfo->contentInfo->mainLocationId;
+        $location = $this->locationHandler->load($mainLocationId);
+        $parentLocation = $this->locationHandler->load($location->parentId);
+        $parentContentInfo = $this->contentHandler->loadContentInfo($parentLocation->contentId);
+
+        return [
+            new Search\Field(
+                'fulltext',
+                $parentContentInfo->name,
+                new Search\FieldType\FullTextField()
+            ),
+        ];
+    }
+}
+```
+
+You index full text data only on the content document, therefore, you would register the service like this:
+
+``` yaml
+my_webinar_app.webinar_event_title_fulltext_field_mapper:
+    class: My\WebinarApp\WebinarEventTitleFulltextFieldMapper
+    arguments:
+        - '@ezpublish.spi.persistence.content_handler'
+        - '@ezpublish.spi.persistence.location_handler'
+    tags:
+        - {name: ezpublish.search.solr.field_mapper.content}
+```
+
+
+!!! caution "Permission issues when using Repository API in document field mappers"
+
+    Document field mappers are low-level and expect to be able to index all content 
+    regardless of current user permissions.
+    If you use PHP API in your custom document field mappers, apply [`sudo()`](../../api/public_php_api.md#using-sudo),
+    or use the Persistence SPI layer as in the example above.
+
+## Custom Search Criteria
+
+To provide support for a custom Search Criterion, do the following.
+
+First, in the `src/Query/Criterion/CameraManufacturerVisitor.php` file, implement `CriterionVisitor`:
```

Comment:
I disagree on the order here. First you should create a Criterion and then a Visitor for it, since the Visitor depends on the criterion.